### PR TITLE
Hot fix gorm issue

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
After the gorm upgrade , migrations of ID column for all tables is being converted into bigserial which is pseudo type in postgres and is not supported in alter command.

There was an issue raised here ragrding the same https://github.com/go-gorm/postgres/issues/65 but the fix mentioned by jinzhu doesn't seem to work even after upgrading the postgres driver for gorm.

Another option mentioned in the issue was to migrate it as bigint which resolves the issue for now but ideally the gorm layer should fix this correctly instead of this hack.

### Testing done:

*  Installed flyte sandbox using the  v0.6.64
*  Registered  cookbook/core and created executions
*  Upgraded flyte to use the flyteadmin's locally tagged image from the PR
* Upgrade went through successfully
* Created another execution with latest flyteadmin which also went through successfully
* Verified from the DB , that ID column changed from int to bigint and it is successfully using the next sequence value 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/repositories/config/migrations.go [hunk 1]
```diff
@@ -330,10 +330,18 @@ var Migrations = []*gormigrate.Migration{
...
+	// Add state to execution model. For any new table migrations please user the following pattern due to following bug
+	// in the postgres gorm layer https://github.com/go-gorm/postgres/issues/65
 	{
 		ID: "2022-01-11-execution-state",
 		Migrate: func(tx *gorm.DB) error {
+			db, err := tx.DB()
+			if err != nil {
+				return err
+			}
+			if _, err = db.Exec(`ALTER TABLE IF EXISTS executions ALTER COLUMN "id" TYPE bigint`); err != nil {
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply pkg/repositories/config/migrations.go:342]: +1 I thought the referenced github issue indicate that just upgrading postgres gorm dep is enough to fix it ?

[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply pkg/repositories/config/migrations.go:342]: @katrogan , I tried looking why thats the case too and went through the gorm code to check how it does the migrations and it only applies the last migration added in the migrations map

https://github.com/go-gormigrate/gormigrate/blob/af859b553353eac85264afd8e7ebd797606a7263/gormigrate.go#L141

Hence i have mentioned that any new migrations that get added need to follow the pattern 

@EngHabu i tried that and if you check go.mod changes,  i have also upgraded to latest postgres gorm but it didn't help hence the change

[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/repositories/config/migrations.go:342]: @pmahindrakar-oss it should apply all migrations up to and including that last one, no? https://github.com/go-gormigrate/gormigrate/blob/af859b553353eac85264afd8e7ebd797606a7263/gormigrate.go#L197,,L204

[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply pkg/repositories/config/migrations.go:345]: Ok i think i misread the code, yes it goes through all the migrations. I was expecting i would run into this issues for other tables aswell which i didn't and hence the confusion

But i see others might run into this issue for other tables based on which was the last state they had migrated to.
Updated the review . Please recheck.

File: pkg/repositories/config/migrations.go [hunk 2]
```diff
@@ -330,18 +330,62 @@ var Migrations = []*gormigrate.Migration{
 		},
 	},
 
-	// Add state to execution model. For any new table migrations please user the following pattern due to following bug
+	// For any new table, Please use the following pattern due to a bug
 	// in the postgres gorm layer https://github.com/go-gorm/postgres/issues/65
 	{
-		ID: "2022-01-11-execution-state",
+		ID: "2022-01-11-id-to-bigint",
```
Comment (katrogan) [inline pkg/repositories/config/migrations.go:336]: can we make this a new, separate migration and add a rollback just in case users face issues?

Comment (pmahindrakar-oss) [reply pkg/repositories/config/migrations.go:346]: Done. its part of two different migration now and also added rollback.

**Comment by pmahindrakar-oss** [comment]
@EngHabu @katrogan please re-review

File: pkg/repositories/config/migrations.go [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+	"fmt"
+
 	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
 	schedulerModels "github.com/flyteorg/flyteadmin/scheduler/repositories/models"
-	gormigrate "github.com/go-gormigrate/gormigrate/v2"
+	"github.com/go-gormigrate/gormigrate/v2"
 	"gorm.io/gorm"
 )
 
+// TODO: add a way to get these list of tables directly from the gorm loaded models
```
Comment (katrogan) [inline pkg/repositories/config/migrations.go:13]: could we do something like https://stackoverflow.com/questions/60976342/using-gorm-to-retrieve-tables-names-from-postgresql to make this programmatic?

Comment (katrogan) [reply pkg/repositories/config/migrations.go:13]: actually this won't affect new tables since those will be automigrated using newer gorm, so a static list is fine.

**Review by katrogan — APPROVED** [review]
